deeppavlov/models/torch_bert/torch_transformers_squad.py

```diff
@@ -124,10 +137,12 @@ def train_on_batch(self, features: List[InputFeatures], y_st: List[List[int]], y
 
         self.optimizer.zero_grad()
 
-        outputs = self.model(input_ids=b_input_ids, attention_mask=b_input_masks,
-                             token_type_ids=b_input_type_ids,
-                             start_positions=b_y_st, end_positions=b_y_end)
-        loss = outputs[0]
+        loss = self.model(input_ids=b_input_ids,
+                          attention_mask=b_input_masks,
+                          token_type_ids=b_input_type_ids,
+                          start_positions=b_y_st,
+                          end_positions=b_y_end,
+                          return_dict=True).loss
```

Comment:
Please, add check if model.forward has all passed arguments (as you did in `torch_transformers_classifier`). Distil* class hasn't `token_type_ids` argument, this code will raise error.

Reply:
Do you checking the accepted_keys property? Like this?
```
tokenized = {key: value for (key, value) in _input.items() if key in self.accepted_keys}
```

Reply:
I mean `if arg in model.forward.__code__.co_varnames`. Yes, it can be implemented using `accepted_keys` property.